lib/customer.rb

```diff
@@ -0,0 +1,25 @@
+
+class Customer
+  ONLY_ONE = 1
+  def initialize(menu = Menu, order = Order, message = Message)
+    @menu = menu.new
+    @order = order.new
+    @message = message.new
+  end
+
+  def look
+    @menu.show
+  end
+
+  def choice(name, amount = ONLY_ONE)
+    @order.add(name, amount)
+  end
+
+  def checkout(my_idea_of_the_cost)
+    message = "You are charging me the incorrect price"
+    raise message if @order.cost != my_idea_of_the_cost
+    puts "You will pay £#{@order.cost}"
+    Pub.new(@order)
```

Comment:
The dependency on `Pub.new()` could be factored into the `initialize` method to break the dependency between `Customer` and `Pub`